### DISCUSSION TITLE
Best practices for error handling?

### DISCUSSION THREAD

**Discussion by mikko-ahonen** [django-components/django-components#480] [Q&A]
How do people handle error signaling from component to the containing view? It would seem natural to use exceptions, but that stops rendering.

**Comment by JuroOravec** [comment]
Good question. I don't have experience from Django, as the Django project I work on is in early development, so error handling wasn't considered yet. So we just let the error propage, and let Django format it. Do not do this, tho.

But you caught my attention with the "but that stops rendering". In React/Vue, there's a concept of ["error boundaries"](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) ([Vue example](https://medium.com/@dillonchanis/handling-errors-in-vue-with-error-boundaries-91f6ead0093b)). It is a place or a "bin" in the template that captures all templating errors inside of it:
```html
<error-boundary>
  <counter />
</error-boundary>
```

How error boundaries work is that when the template inside renders successfully, it's as if the error boundary wasn't there. But if there is an error during render, then the content of `error-boundary` is replaced with the error message (or optionally some fallback content). And the rest of the template is allowed to continue rendering.

Is that what you're thinking of?

After a quick google search I haven't found a Django version of error boundary.

**Comment by mikko-ahonen** [comment]
Yes, something like that. One typical scenario:
* I have a component to change let's say some date of a model entity
* Backend validation for the input may fail
* Typically validation failure is communicated in Django with ValidationError
* I would like to raise ValidationError in component get/post method and get the message propagated to the user

**Reply by JuroOravec** [reply]
Ok, that might be actually something different. I suppose you're using Django's forms for the validation? Don't have much experience with them, but there should be a way to report the validation errors.
